.github/workflows/ci.yml

```diff
@@ -35,28 +38,40 @@ jobs:
       matrix:
         os: [ubuntu-latest]
         rust: [stable, 1.45.0, beta]
+        args: ["", "--all-features"] 
     steps:
     - name: Checkout repository
       uses: actions/checkout@v1
     - name: Install Rust
       uses: hecrj/setup-rust-action@v1
       with:
         rust-version: ${{ matrix.rust }}
-    - run: cargo build --verbose
-    - run: cargo test --verbose 2>&1
-  allfeatures:
-    name: allfeatures
+    - run: cargo build --verbose ${{ matrix.args }}
+    - run: cargo test --verbose ${{ matrix.args }} 2>&1
+
+  coverage:
+    name: coverage
+    needs: build
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
         os: [ubuntu-latest]
-        rust: [stable, 1.45.0, beta]
+        rust: [stable]
+        args: ["", "--all-features"] 
     steps:
     - name: Checkout repository
       uses: actions/checkout@v1
     - name: Install Rust
       uses: hecrj/setup-rust-action@v1
       with:
         rust-version: ${{ matrix.rust }}
-    - run: cargo build --verbose --all-features
-    - run: cargo test --verbose --all-features 2>&1
+    - run: cargo install cargo-tarpaulin
```

Comment:
Perhaps use https://github.com/cypress-io/github-action here instead of compiling tarpauling via `cargo install`?

Reply:
This isn't going to work as the action runs the tests (it doesn't just pull in the Cypress test runner Docker image) and currently the Cypress test runner is invoked by Rust integration test code (not by the GH workflow) which also does pre and post steps around the invocation of the Cypress runner (e.g. start and stop Krill, start and stop the mock OpenID Connect provider, inspect the Krill event history after the test, etc).

Reply:
I won't investigate this at the moment as the win isn't big enough to warrant the additional time to test the tarpaulin action and to work out where it writes the output report and to upload it correctly. Currently the workflow run for all features against the v0.8.1-bis branch takes 28 minutes of which only 3 minutes is spent installing cargo tarpaulin.